src/internal/observable/from.ts

```diff
@@ -119,34 +119,130 @@ export function from<O extends ObservableInput<any>>(input: O, scheduler: Schedu
  * @owner Observable
  */
 export function from<T>(input: ObservableInput<T>, scheduler?: SchedulerLike): Observable<T> {
-  if (!scheduler) {
-    if (input instanceof Observable) {
-      return input;
+  return scheduler ? scheduled(input, scheduler) : innerFrom(input);
+}
+
+// TODO: Use this throughout the library, rather than the `from` above, to avoid
+// the unnecessary scheduling check and reduce bundled sizes of operators that use `from`.
+// TODO: Eventually, this just becomes `from`, as we don't have the deprecated scheduled path anymore.
+export function innerFrom<T>(input: ObservableInput<T>) {
+  if (input instanceof Observable) {
+    return input;
+  }
+  if (input != null) {
+    if (isInteropObservable(input)) {
+      return fromInteropObservable(input);
+    }
+    if (isArrayLike(input)) {
+      return fromArrayLike(input);
+    }
+    if (isPromise(input)) {
+      return fromPromise(input);
+    }
+    if (isIterable(input)) {
+      return fromIterable(input as any);
+    }
+    if (isAsyncIterable(input)) {
+      return fromAsyncIterable(input as any);
```

Comment:
Same `fromAsyncIterable` and `fromIterable`?

Reply:
Yeah... although looking at those, I'm going to need to reverse their order I think. 🤔